### PR TITLE
Upgrade dependencies and improve Readme.md

### PR DESCRIPTION
**Pull request by patbaumgartner**
- Updated Maven & Gradle Plugins
- Updated Maven & Gradle Wrapper to the current distribution
- Updated the DB containers dependency
- Updated webjars
- Improved the Readme

### CODE REVIEW

**Comment by patbaumgartner** [comment]
Hi @dsyer some of my changes were not addressed by the Spring Boot 3.2 upgrade. Therefore, I created a new one with the latest adjustments.

File: readme.md [hunk 1]
```diff
@@ -151,24 +151,12 @@ Here is a list of them:
...
+The [issue tracker](https://github.com/spring-projects/spring-petclinic/issues) is the preferred channel for bug reports, feature requests and submitting pull requests.
 
 For pull requests, editor preferences are available in the [editor config](.editorconfig) for easy use in common text editors. Read more and download plugins at <https://editorconfig.org>. If you have not previously done so, please fill out and submit the [Contributor License Agreement](https://cla.pivotal.io/sign/spring).
 
-# License
+## License
 
 The Spring PetClinic sample application is released under version 2.0 of the [Apache License](https://www.apache.org/licenses/LICENSE-2.0).
-
-[spring-petclinic]: https://github.com/spring-projects/spring-petclinic
```
Comment (dsyer) [inline readme.md:165]: Did we need to delete all these links? They seem useful to me.

**Review by dsyer — COMMENTED** [review]
Tiny question in comments. Thanks for all the effort.

**Comment by patbaumgartner** [comment]
Hi @dsyer

Unfortunately the links are not visible rendered on GitHubs website or
anywhere else. It's incorrect MD which is ignored.

Also the list is incomplete, because the spring pet clinic GH organization
contains way more repos than those.

The GH organization is linked in the section "Interesting Spring Petclinic
branches and forks". These links should also go to that section or get
their own title.

It's too much of a mess, so I thought it's better to remove them.

**Pull request merged**